virtcontainers/sandbox.go

```diff
@@ -1201,6 +1207,42 @@ func (s *Sandbox) CreateContainer(contConfig ContainerConfig) (VCContainer, erro
 	return c, nil
 }
 
+func (s *Sandbox) creditContainerVCPU(num uint32, credit bool) (uint32, error) {
+	if credit {
+		if s.state.FreeStaticCPU == 0 {
+			return num, nil
+		}
+
+		if num <= s.state.FreeStaticCPU {
+			s.state.FreeStaticCPU -= num
```

Comment:
This function is changing the state, but if the subsequent call to `c.sandbox.hypervisor.hotplug[Add|Remove]Device()` fails, won't the state then be incorrect?

Reply:
@bergwolf Same question. I think you would need to call this function again with reverse credit flag?

Reply:
The state is only saved on disk when hotplug succeeds.